### ISSUE TITLE
When the hyperparameters importance plot is generated it does not label the parameters by its actual name

### ISSUE THREAD

**Issue by RUMCS** [optuna/optuna#5162]
### Expected behavior

When the hyperparameters importance plot is generated, it does not label the parameters by their actual name. For instance, it produces a histogram where the parameters are labeled as 0,1,2,3,4 instead of the actual ones defined in the code, as: learning rate, batch size, number of epochs, etc... 

### Environment

- Optuna version: 3.4.0
- Python version: 3.8.10
- OS: Linux Debian
- (Optional) Other libraries and their versions: Matplotlib

### Error messages, stack traces, or logs

```shell
no errors
```

### Steps to reproduce

 1.  Run the code
 2. 
    study = optuna.create_study(
    direction='maximize',
    study_name='myresearch_study', 
    storage='postgresql://XXXX.dbpass@localhost/optuna_study',  # Use your PostgreSQL credentials and database name
    load_if_exists=True)

    n_trials = args.trials ## set num of trials
    study.optimize(lambda trial: optimize_hyperparameters(trial, X_train, y_train, device, args), n_trials) 

    # Best hyperparameters
    print("Best hyperparameters: ", study.best_params)
...
    # Plotting parameter importances
    ax = plot_param_importances(study)
    fig = ax.figure
    fig.set_size_inches(8, 6)  # Set the dimensions after getting the figure
    fig.savefig("param_importances.png")

...

### Additional context (optional)

![image](https://github.com/optuna/optuna/assets/12820300/b032d84a-1c79-4be4-b028-53f56e3f5bf6)

**Comment by nzw0301** [comment]
Could you share the figure by running the code example in 
https://optuna.readthedocs.io/en/stable/reference/visualization/generated/optuna.visualization.matplotlib.plot_param_importances.html#optuna.visualization.matplotlib.plot_param_importances? This is because your script does not contain the objective.

**Comment by RUMCS** [comment]
Thanks for now I am including the objective function ...
